src/moi.jl

```diff
@@ -406,13 +413,186 @@ function build_bilevel(
+            upper_obj_func_quad_terms = MOI.get(upper, MOI.ObjectiveFunction{MOI.get(upper, MOI.ObjectiveFunctionType())}()).quadratic_terms
+            linearizations = nothing
+            m_objective = MOI.get(m, MOI.ObjectiveFunction{MOI.get(m, MOI.ObjectiveFunctionType())}())
+            bilinear_upper_quad_term_to_m_quad_term = Dict{MOI.ScalarQuadraticTerm, MOI.ScalarQuadraticTerm}()
+
+            for term in m_objective.quadratic_terms
+                mset = Set([term.variable_index_1, term.variable_index_2])
+                for upper_term in upper_obj_func_quad_terms
+                    uset = Set([
+                        upper_to_m_idxmap[upper_term.variable_index_1],
+                        upper_to_m_idxmap[upper_term.variable_index_2]
+                    ])
+                    if uset == mset
+                        bilinear_upper_quad_term_to_m_quad_term[upper_term] = term
+                    end
+                end
+            end
+
+            # TODO check for integer x * continuous y, for now assuming continuous x conditions
+
+            J_U = Int[]
+            N_U = Int[]
+            for (upper_var, lower_con) in upper_var_to_lower_ctr  # equivalent to set A with pairs (j,n) : A_jn ≠ 0
+                j = lower_con.value
+                n = bilinear_upper_dual_to_lower_primal[upper_var].value
+                if !(n in keys(bilinear_upper_dual_to_lower_primal)) continue end  # user defined DualOf but did not use it in UL objective
+                rows, cols = find_connected_rows_cols(V, j, n, skip_1st_col_check=!(isempty(AB_N)))
+                push!(J_U, rows...)
+                push!(N_U, cols...)
+            end
+            #= 
+                Case without x_m * y_n in LL objective for all y_n in A_N (set of bilinear UL objective terms of form λ_j * y_n)
+            =#
+            if isempty(AB_N)
+                @debug("set AB_N is empty")
+
+                conditions_passed = check_empty_AB_N_conditions(J_U, U, N_U, B)
+
+                if conditions_passed
+                    linearizations = linear_terms_for_empty_AB(
+                        lower,
+                        upper_var_to_lower_ctr,
+                        bilinear_upper_dual_to_lower_primal,
+                        V,
+                        w,
+                        bilinear_upper_dual_to_quad_term,
+                        lower_obj_terms,
+                        lower_to_m_idxmap,
+                        lower_primal_dual_map,
+                        lower_dual_idxmap
+                    )
+                    if isnothing(linearizations)
+                        @warn("Unable to linearize bilinear terms due to underdertermined system of equations. Skipping linearization process.")
+                    end
+                else
+                    @warn("Required conditions for linearization not met. Skipping linearization process.")
```

Comment:
can you say which? referencing the paper seems fine.

Reply:
Yes the @debug statements tell users which conditions were not meant. Would you prefer that the unmet conditions are collected into this @warn statement?

Reply:
in this case I meant, lets add more details about what condition is failing to be met, keep as warning.

Reply:
addressed in b02f8bc